### PR TITLE
[WFLY-14721] Add tests to XTS and transactions(JTA) subsystem for attributes that allow expression

### PR DESCRIPTION
**Pull request by petrberan**
https://issues.redhat.com/browse/WFLY-14721

### CODE REVIEW

File: transactions/src/test/resources/org/jboss/as/txn/subsystem/full-expressions.xml [hunk 1]
```diff
@@ -16,5 +16,5 @@
             <xid-location name="${test.table:fooTable}" batch-size="${test.batch:10}" immediate-cleanup="${test.immediate:false}"/>
         </commit-markable-resource>
     </commit-markable-resources>
-    
+    <client stale-transaction-time="${test.stale-transaction-time:600}"/>
```
Comment (mmusgrov) [inline transactions/src/test/resources/org/jboss/as/txn/subsystem/full-expressions.xml:19]: That is the default value but I guess it does no harm to specify it explicitly.

**Review by mmusgrov — APPROVED** [review]
The changes look fine, thanks very much.

**Comment by jmfinelli** [comment]
Hi @petrberan. As [WFLY-14721](https://issues.redhat.com/browse/WFLY-14721) looks a bit out-of-date, could you please add a comment in that JIRA to explain why the modifications in this PR only partially address the description of the issue? Thanks

**Comment by jmfinelli** [comment]
Thanks @petrberan. LGTM! I'll leave the approval to @mmusgrov

**Comment by mmusgrov** [comment]
+1 the two changes still look OK

**Comment by mmusgrov** [comment]
> @mmusgrov Parsing of the XML is `TransactionSubsystemxxParser` responsibility (the xx standing for 60/50/40/... depending on the version). The syntax error was noticed by me after a week of figuring out the problem with the `jndi-name` attribute - once the `socket-binding` was parsed into Model Node it was missing the usual "expression" prefix.
> 

Nice work.

> Perhaps a build regex check resulting in a warning if an incomplete expression was found would be a good addition

We need a regex expert, do we have any ;)

**Pull request merged**